crates/autopilot/src/database/on_settlement_event_updater.rs

```diff
@@ -60,6 +60,18 @@ impl super::Postgres {
             )
             .await
             .context("insert_settlement_observations")?;
+
+            // delete auction prices for auction_id and all auctions before it
+
+            // TODO dont delete all auctions, instead:
```

Comment:
Should I do this? I think I should

Reply:
I think we probably don't have to do `1`. Prices that don't get deleted because we didn't do `1` will get deleted when we index the next auction so I think having additional code is not worth it.

Reply:
> I am not aware of this. Can you show me the code where it happens?

I thought this is what happens in the code right after this comment block. Seems like I misunderstood what's actually happening there.

Reply:
> then we can save the auction prices before submission and there is no need for this code.

I think there already is no need for this code. We can save prices for all tokens for which there are orders (from the `order_executions` vector).